Source/DataModels/Components/ComponentMeshCollider.cpp

```diff
@@ -0,0 +1,195 @@
+	GetPointsGivenDirection(usedPoints, PositionPoints, direction);
+	float3 movementDirection = GetMovementGivenDirection(PositionPoints, direction);
+
+	if (!IsColliding(usedPoints, movementDirection, size, hit, stepSize))
+	{
+		position += movementDirection * size;
+		trans->SetPosition(position);
+
+		trans->UpdateTransformMatrices();
+		return true;
+	}
+	return false;
+}
+
+bool ComponentMeshCollider::StepsMove(int steps, Direction direction, float distance, RaycastHit& hit, float stepSize)
+{
+	float sectionMove = distance / (float)steps;
+
+	ComponentTransform* trans = static_cast<ComponentTransform*>(owner->GetComponent(ComponentType::TRANSFORM));
+	float3 position = trans->GetPosition();
+
+	math::vec points[8];
+	trans->GetObjectOBB().GetCornerPoints(points);
+	std::vector<float3> PositionPoints = { points[0], points[1],  points[2],  points[3],  points[4],  points[5],  points[6],  points[7] };
+	std::vector<float3> usedPoints;
+	GetPointsGivenDirection(usedPoints, PositionPoints, direction);
+	float3 movementDirection = GetMovementGivenDirection(PositionPoints, direction);
+	for (int i = steps; i > 0; i--)
+	{
+		if(!IsColliding(usedPoints, movementDirection, sectionMove*(float)i, hit, stepSize))
+		{
+			position += movementDirection * sectionMove * (float)i;
+			trans->SetPosition(position);
+
+			trans->UpdateTransformMatrices();
+			return true;
+		}
+	}
+
+	return false;
+}
+
+bool ComponentMeshCollider::IsColliding(std::vector<float3>& startingPoints, float3 direction, float size, RaycastHit& hit, float stepSize) const
+{
+	std::vector<float3> points;
+	GetMinMaxPoints(startingPoints, points, stepSize);
+
+	for (float3 point : points)
+	{
+		Ray ray(point, direction);
+		LineSegment line(ray, size);
+		bool hasHit = Physics::Raycast(line, hit);
+
+		if (hasHit) {
+			return true;
+		}
+
+	}
+
+	return false;
+}
+
+void ComponentMeshCollider::GetMinMaxPoints(std::vector<float3>& startingPoints, std::vector<float3>& points, float stepSize) const
+{
+	points = startingPoints;
+
+	points.push_back(startingPoints[0] + (startingPoints[1] - startingPoints[0]) / 2);
+	points.push_back(startingPoints[0] + (startingPoints[2] - startingPoints[0]) / 2);
+	points.push_back(startingPoints[3] - (startingPoints[3] - startingPoints[2]) / 2);
+	points.push_back(startingPoints[3] - (startingPoints[3] - startingPoints[1]) / 2);
+	points.push_back(startingPoints[0] + (startingPoints[3] - startingPoints[0]) / 2);
+
+	//Add stepSize to the top 3 down;
+	if (stepSize != 0)
+	{
+		float minStep = points[0].y;
+		for (float3 point : points)
+		{
+			if (minStep > point.y)
+			{
+				minStep = point.y;
+			}
+		}
+		minStep += stepSize;
+		for (int i = 0; i < points.size(); i++)
+		{
+			if (minStep > points[i].y)
+			{
+				points[i] = float3(points[i].x, minStep, points[i].z);
+			}
+		}
+	}
+}
+
+
+void ComponentMeshCollider::GetPointsGivenDirection(std::vector<float3>& newPoints, std::vector<float3>& points, Direction direction) const

```

Comment:
same as before, also this is only used in .cpp so could be in private not public
```suggestion
std::vector<float3> ComponentMeshCollider::GetPointsGivenDirection(std::vector<float3>& points, Direction direction) const

```